### PR TITLE
HPCC-22602 Initial check-in Docker build files

### PR DESCRIPTION
**Pull request by xwang2713**
HPCC Platform Community Version
  ubuntu (bionic)/
  -  base/<version>/Dockerfile: HPCC Platform Community version prequisites
     Docker Hub:  hpccsystems/hpcc-base
  -  platform/<version>/Dockerfile  HPCC Platform Community Docker build file
     Docker Hub:  hpccsystems/platform
  -  clienttools/<version>Dockerfile: HPCC Clienttools Community Docker build file
     Docker Hub: hpccsystems/clienttools
Development and custom Build
  <bionic|disco|eoan>/
  -   base/Dockerfile: HPCC Platform community version prequisites
  -   dev/<version>/Dockerfile: Build Server Docker build file
  -   platform/<version>/Dockerfile: Docker build file for compiling and building HPCC Platform image
  -   clienttools/<version>/Dockerfile: Docker build file for compiling and building HPCC Clienttools image

  gcc/<7|8|9>/
  -   base/Dockerfile: HPCC Platform community version prequisites
  -   dev/<version>/Dockerfile: Build Server Docker build file
  -   platform/<version>/Dockerfile: Docker build file for compiling and building HPCC Platform image

The final HPCC Docker build files have "ARG" for "version" so they don't need update these source files for every build. The "version" will be provided in "docker build" command.

@rpastrana @gfortil please review 
@g-pan please review README.md:  
    containers/docker/README.md,  containers/docker/ubuntu/platform/README.md and containers/docker/ubuntu/clienttools/README.md

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: containers/docker/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# HPCC Platform Community Version
+  ubuntu (bionic)/
+  -  base/<version>/Dockerfile: HPCC Platform Community version prequisites
```
Comment (g-pan) [inline containers/docker/README.md]: HPCC Systems platform community

File: containers/docker/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# HPCC Platform Community Version
```
Comment (g-pan) [inline containers/docker/README.md]: HPCC Systems(r) Platform Community Version

File: containers/docker/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  ubuntu (bionic)/
+  -  base/<version>/Dockerfile: HPCC Platform Community version prequisites
+     Docker Hub:  hpccsystems/hpcc-base
+  -  platform/<version>/Dockerfile  HPCC Platform Community Docker build file
```
Comment (g-pan) [inline containers/docker/README.md]: HPCC Systems platform community

File: containers/docker/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+     Docker Hub:  hpccsystems/platform
+  -  clienttools/<version>Dockerfile: HPCC Clienttools Community Docker build file
```
Comment (g-pan) [inline containers/docker/README.md]: HPCC Systems client tools community

File: containers/docker/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+     Docker Hub: hpccsystems/clienttools
+
+## How to buid
```
Comment (g-pan) [inline containers/docker/README.md]: How to build

File: containers/docker/README.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+Create HPCC Platform Docker image :
```
Comment (g-pan) [inline containers/docker/README.md]: Create HPCC Systems platform Docker image:

File: containers/docker/README.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+sudo docker build -t hpccsystems/platform:7.4.8-1 --build-arg version=7.4.8-1 .
+```
+
+
+# HPCC Platform Internal Version
+  Reference LN container/docker/
+
+# Development and custom Build
+  <bionic|disco|eoan>/
+  -   base/Dockerfile: HPCC Platform community version prequisites
```
Comment (g-pan) [inline containers/docker/README.md]: HPCC Systems platform community

File: containers/docker/README.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  -   dev/<version>/Dockerfile: Build Server Docker build file
+  -   platform/<version>/Dockerfile: Docker build file for compiling and building HPCC Platform image
```
Comment (g-pan) [inline containers/docker/README.md]: compiling and building the HPCC Systems platform image

File: containers/docker/README.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  -   clienttools/<version>/Dockerfile: Docker build file for compiling and building HPCC Clienttools image
```
Comment (g-pan) [inline containers/docker/README.md]: HPCC Systems client tools image

File: containers/docker/README.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+  gcc/<7|8|9>/
+  -   base/Dockerfile: HPCC Platform community version prequisites
```
Comment (g-pan) [inline containers/docker/README.md]: HPCC Systems platform community version prerequisites

File: containers/docker/README.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+  -   dev/<version>/Dockerfile: Build Server Docker build file
+  -   platform/<version>/Dockerfile: Docker build file for compiling and building HPCC Platform image
```
Comment (g-pan) [inline containers/docker/README.md]: HPCC Systems platform image

File: containers/docker/README.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+## How to build
+  The compiling and building Docker build file has "ARG" for various input parameter.
```
Comment (g-pan) [inline containers/docker/README.md]: Compiling and building the Docker build file has an "ARG" for various input parameters.

**Review by g-pan — COMMENTED** [review]
See comments inline.

**Comment by rpastrana** [comment]
The directory structure should be restructured as we discussed offline. @xwang2713

**Comment by richardkchapman** [comment]
Closing based on the above comment

**Pull request closed without merging**